src/jit/codegencommon.cpp

```diff
@@ -4282,7 +4282,14 @@ void CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg, bool* pXtraRegClobbere
 #endif // defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
         {
             // Bingo - add it to our table
-            regArgNum = genMapRegNumToRegArgNum(varDsc->lvArgReg, regType);
+            if (!compiler->opts.compUseSoftFP)
+            {
+                regArgNum = genMapRegNumToRegArgNum(varDsc->lvArgReg, regType);
+            }
+            else
+            {
+                regArgNum = genMapIntRegNumToRegArgNum(varDsc->lvArgReg);
+            }
 
```

Comment:
Can you explain why this is necessary? In particular, this function (genFnPrologCalleeRegArgs) is only called if `regState->rsCalleeRegArgMaskLiveIn` is non-empty. For the `compUseSoftFP` case, shouldn't the floating-point register file have an empty set for `regState->rsCalleeRegArgMaskLiveIn`?